platform/initialize.go

```diff
@@ -0,0 +1,144 @@
+package platform
+
+import (
+	"context"
+	"github.com/kiegroup/container-builder/api"
+	"github.com/kiegroup/container-builder/client"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/builder"
+	"github.com/kiegroup/kogito-serverless-operator/constants"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewInitializeAction returns a action that initializes the platform configuration when not provided by the user.
+func NewInitializeAction() Action {
+	return &initializeAction{}
+}
+
+type initializeAction struct {
+	baseAction
+}
+
+func (action *initializeAction) Name() string {
+	return "initialize"
+}
+
+func (action *initializeAction) CanHandle(platform *v08.KogitoServerlessPlatform) bool {
+	return platform.Status.Phase == "" || platform.Status.Phase == v08.PlatformPhaseDuplicate
+}
+
+func (action *initializeAction) Handle(ctx context.Context, platform *v08.KogitoServerlessPlatform) (*v08.KogitoServerlessPlatform, error) {
+	duplicate, err := action.isPrimaryDuplicate(ctx, platform)
+	if err != nil {
+		return nil, err
+	}
+	if duplicate {
+		// another platform already present in the namespace
+		if platform.Status.Phase != v08.PlatformPhaseDuplicate {
+			platform := platform.DeepCopy()
+			platform.Status.Phase = v08.PlatformPhaseDuplicate
+
+			return platform, nil
+		}
+
+		return nil, nil
+	}
+
+	if err = ConfigureDefaults(ctx, action.client, platform, true); err != nil {
+		return nil, err
+	}
+	// nolint: staticcheck
+	if platform.Status.BuildPlatform.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
+		cacheEnabled := platform.Status.BuildPlatform.IsOptionEnabled(builder.KanikoBuildCacheEnabled)
```

Comment:
it is not used on other places, maybe remove var assignment?

Reply:
yes, I did it